x-pack/plugin/esql/compute/src/main/generated-src/org/elasticsearch/compute/aggregation/RateDoubleAggregator.java

```diff
@@ -0,0 +1,276 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.compute.aggregation;
+
+import org.apache.lucene.util.Accountable;
+import org.apache.lucene.util.RamUsageEstimator;
+import org.elasticsearch.common.util.BigArrays;
+import org.elasticsearch.common.util.ObjectArray;
+import org.elasticsearch.compute.ann.GroupingAggregator;
+import org.elasticsearch.compute.ann.IntermediateState;
+import org.elasticsearch.compute.data.Block;
+import org.elasticsearch.compute.data.BlockFactory;
+import org.elasticsearch.compute.data.DoubleBlock;
+import org.elasticsearch.compute.data.DoubleVector;
+import org.elasticsearch.compute.data.IntVector;
+import org.elasticsearch.compute.data.LongBlock;
+import org.elasticsearch.compute.operator.DriverContext;
+import org.elasticsearch.core.Releasable;
+import org.elasticsearch.core.Releasables;
+
+/**
+ * A rate grouping aggregation definition for double.
+ * This class is generated. Edit `X-RateAggregator.java.st` instead.
+ */
+@GroupingAggregator(
+    includeTimestamps = true,
+    value = {
+        @IntermediateState(name = "timestamp", type = "LONG_BLOCK"),
+        @IntermediateState(name = "value", type = "DOUBLE_BLOCK"),
+        @IntermediateState(name = "compensation", type = "DOUBLE") }
+)
+public class RateDoubleAggregator {
```

Comment:
I'm somehow totally unsurpised we've started to use one code generation tool to kick off another code generation tool.

Reply:
Maybe the next time we find something interesting to do there. I'm not sure it's worth doing without some kind of excuse.